### PR TITLE
picoprobe-udev-rules: init at unstable-2023-01-31

### PR DESCRIPTION
**Pull request by MGlolenstine**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/picoprobe-udev-rules/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+## Usage
+# In NixOS, simply add this package to services.udev.packages:
+#   services.udev.packages = [ pkgs.picoprobe-udev-rules ];
+
+stdenv.mkDerivation rec {
+  pname = "picoprobe-udev-rules";
+  version = "0.0.1";
+
+  src = fetchurl {
+    url = "https://probe.rs/files/69-probe-rs.rules";
```
Comment (NickCao) [inline pkgs/os-specific/linux/picoprobe-udev-rules/default.nix:12]: Aren't there some *permanent* url whose content won't change over time?

Comment (MGlolenstine) [reply pkgs/os-specific/linux/picoprobe-udev-rules/default.nix:12]: I don't think so, at least not in the official repository.

Comment (NickCao) [reply pkgs/os-specific/linux/picoprobe-udev-rules/default.nix:12]: Found it here https://github.com/probe-rs/webpage/blob/master/static/files/69-probe-rs.rules

Comment (MGlolenstine) [reply pkgs/os-specific/linux/picoprobe-udev-rules/default.nix:12]: Oh, I'll port it to the latest commit from that repo, thanks!

Comment (NickCao) [reply pkgs/os-specific/linux/picoprobe-udev-rules/default.nix:12]: And just set the version to `unstable-YYYY-MM-DD`.

**Review by NickCao — APPROVED** [review]

**Pull request merged**